code/SNIDsn.py

```diff
@@ -410,59 +410,86 @@ def loadSNIDlnw(self, lnwfile):
         with open(lnwfile) as lnw:
             lines = lnw.readlines()
             lnw.close()
-        header_line = lines[0].strip()
-        header_items = header_line.split()
-        header = dict()
-        header['Nspec'] = int(header_items[0])
-        header['Nbins'] = int(header_items[1])
-        header['WvlStart'] = float(header_items[2])
-        header['WvlEnd'] = float(header_items[3])
-        header['SplineKnots'] = int(header_items[4])
-        header['SN'] = header_items[5]
-        header['dm15'] = float(header_items[6])
-        header['TypeStr'] = header_items[7]
-        header['TypeInt'] = int(header_items[8])
-        header['SubTypeInt'] = int(header_items[9])
-        self.header = header
-
-        tp, subtp = getType(header['TypeInt'], header['SubTypeInt'])
-        self.type = tp
-        self.subtype = subtp
-
-        phase_line_ind = len(lines) - self.header['Nbins'] - 1
-        phase_items = lines[phase_line_ind].strip().split()
-        self.phaseType = int(phase_items[0])
-        phases = np.array([float(ph) for ph in phase_items[1:]])
-        self.phases = phases
-
-        wvl = np.loadtxt(lnwfile, skiprows=phase_line_ind + 1, usecols=0)
-        self.wavelengths = wvl
-        lnwdtype = []
-        colnames = []
-        for ph in self.phases:
-            colname = 'Ph'+str(ph)
-            if colname in colnames:
-                colname = colname + 'v1'
-            count = 2
-            while(colname in colnames):
-                colname = colname[0:-2] + 'v'+str(count)
-                count = count + 1
-            colnames.append(colname)
-            dt = (colname, 'f4')
-            lnwdtype.append(dt)
-        #lnwdtype = [('Ph'+str(ph), 'f4') for ph in self.phases]
-        #print lines[phase_line_ind+1]
-        data = np.loadtxt(lnwfile, dtype=lnwdtype, skiprows=phase_line_ind + 1, usecols=range(1,len(self.phases) + 1))
-        self.data = data
-
-        continuumcols = len(lines[1].strip().split())
-        continuum = np.ndarray((phase_line_ind - 1,continuumcols))
-        for ind in np.arange(1,phase_line_ind - 0):
-            cont_line = lines[ind].strip().split()
-            #print cont_line
-            continuum[ind - 1] = np.array([float(x) for x in cont_line])
-        self.continuum = continuum
-        return
+        if len(header_lines) == 10: #if file matches tutorial data format
```

Comment:
This is hard for users to understand.  Use '.lnw' for SNID templates. If suffix of the user specified file isn't lnw, then check to see if the file has two columns, one for wavelength and one for flux.  If it does, make a SNIDsn object, otherwise, raise an assert.

Reply:
look up Python assertions.

Reply:
Maybe allow ascii files with at least 2 columns. Require a header line first that names the columns.

Reply:
your new function should have a phaseType argument for the user to specify whether phases are defined relative to max or not.